C++ interop: add a simple unit test for "using enum"

neundorf opened the pull request:
This unit test uses "using UEnum = Enum;" and imports that into Carbon.

This is my very first try at contributing something to carbon, I'm looking forward to your feedback.
The test is very basic. 
What other cases should it test ?
What other comments do you have ?

neundorf commented:
The test currently simply tests one using enum on global scope.
Should I add an enum inside a class ? An enum class inside a class ? Other variations ?

zygoloid commented:
If this is for the "using enum support" item in the toolchain tasks document, that's talking about the [C++20 `using enum` feature](https://en.cppreference.com/w/cpp/language/enum.html#:~:text=since%20C%2B%2B17%29-,using%20enum%20declaration,-using%20enum%20using):
```c++
enum class A { x, y };
// Cannot name `A::x` as simply `x` here.
using enum A;
// Can now name `A::x` as simply `x`.
A a = x;
```
The `using B = A;` syntax is an alternative syntax for `typedef` declarations, and we have tests for that in `toolchain/check/testdata/interop/cpp/typedef.carbon` already.

neundorf commented:
> If this is for the "using enum support" item in the toolchain tasks document, that's talking about the [C++20 `using enum` feature](https://en.cppreference.com/w/cpp/language/enum.html#:~:text=since%20C%2B%2B17%29-,using%20enum%20declaration,-using%20enum%20using):
> 
> ```c++
> enum class A { x, y };
> // Cannot name `A::x` as simply `x` here.
> using enum A;
> // Can now name `A::x` as simply `x`.
> A a = x;
> ```
> 
> The `using B = A;` syntax is an alternative syntax for `typedef` declarations, and we have tests for that in `toolchain/check/testdata/interop/cpp/typedef.carbon` already.

thanks, I'll update it.

neundorf commented:
> > If this is for the "using enum support" item in the toolchain tasks document, that's talking about the [C++20 `using enum` feature](https://en.cppreference.com/w/cpp/language/enum.html#:~:text=since%20C%2B%2B17%29-,using%20enum%20declaration,-using%20enum%20using):
> > ```c++
> > enum class A { x, y };
> > // Cannot name `A::x` as simply `x` here.
> > using enum A;
> > // Can now name `A::x` as simply `x`.
> > A a = x;
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The `using B = A;` syntax is an alternative syntax for `typedef` declarations, and we have tests for that in `toolchain/check/testdata/interop/cpp/typedef.carbon` already.
> 
> thanks, I'll update it.

Ok, I'm not sure what should be tested here actually.
Should ` using enum Foo;`  actually have an effect when this is imported into Carbon ?
Should it then be possible to write
```c++
let a : cpp.Foo = SomeValue; 
```

(instead of cpp.Foo.SomeValue) ?

zygoloid commented:
> Should ` using enum Foo;` actually have an effect when this is imported into Carbon ? Should it then be possible to write
> 
> ```c++
> let a : cpp.Foo = SomeValue; 
> ```
> 
> (instead of cpp.Foo.SomeValue) ?

It should be possible to write
```carbon
let a: Cpp.Foo = Cpp.SomeValue;
```
instead of `Cpp.Foo.SomeValue`.

neundorf committed:
C++ interop: add a simple unit test for "using enum"

This unit test uses the C++20 "using enum SomeEnum;" and
imports that into Carbon.

zygoloid reviewed (approved):
Thanks!

neundorf commented:
@zygoloid does the "Carbon Language - Toolchain tasks" document have to be updated ? Who is taking care of this ?

zygoloid commented:
Thanks for the reminder, done.
